Call coregrind's allocator from krabcake

bryangarza opened the pull request:
This patch creates a global allocator that uses the implementation provided by Valgrind (which lives in coregrind).

This allows us to pull in the `alloc` crate.

bryangarza committed:
Call coregrind's allocator from krabcake

This patch creates a global allocator that uses the implementation
provided by Valgrind (which lives in coregrind).

This allows us to pull in the `alloc` crate.

pnkfelix commented on krabcake/rs_hello/src/lib.rs:
```diff
@@ -1,9 +1,51 @@
+
+unsafe impl GlobalAlloc for ValgrindAllocator {
+    unsafe fn alloc(&self, layout: Layout) -> *mut u8 {
+        let cc: &[u8] = b"cc"; // TODO: Set a unique identifier here?
```
bryan: Does this need to be null-terminated? I.e. `b"cc\0"` ? IIRC Rust won't do that for you and I cannot guess how vgPlain_malloc is going to figure out how long the string is otherwise.

pnkfelix replied:
```suggestion
        let cc: &[u8] = b"cc\0"; // TODO: Set a unique identifier here?
```

pnkfelix commented:
~~r=me once @bryangarza has either added a null-terminator or convinced me its not necessary. :)~~

(I just did it myself, because I want to rebase atop these changes.)

pnkfelix committed:
Update krabcake/rs_hello/src/lib.rs

null-terminate the bytestring that ends up getting passed along into Valgrind's C-based malloc method.
